fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java

```diff
@@ -514,5 +576,121 @@ public Void visitDropTableStatement(DropTableStmt statement, ConnectContext sess
+        }
+
+        @Override
+        public Void visitAdminShowReplicaStatusStatement(AdminShowReplicaStatusStmt statement, ConnectContext context) {
+            return checkStmtOperatePrivilege(context);
+        }
+
+        @Override
+        public Void visitAdminRepairTableStatement(AdminRepairTableStmt statement, ConnectContext context) {
+            return checkStmtOperatePrivilege(context);
+        }
+
+        @Override
+        public Void visitAdminCancelRepairTableStatement(AdminCancelRepairTableStmt statement, ConnectContext context) {
+            return checkStmtOperatePrivilege(context);
+        }
+
+        @Override
+        public Void visitAdminCheckTabletsStatement(AdminCheckTabletsStmt statement, ConnectContext context) {
+            return checkStmtOperatePrivilege(context);
+        }
+
+        @Override
+        public Void visitKillStatement(KillStmt statement, ConnectContext context) {
+            return checkStmtOperatePrivilege(context);
+        }
+
+        @Override
+        public Void visitAlterSystemStatement(AlterSystemStmt statement, ConnectContext context) {
+            return checkStmtNodePrivilege(context);
+        }
+
+        @Override
+        public Void visitCancelAlterSystemStatement(CancelAlterSystemStmt statement, ConnectContext context) {
+            return checkStmtNodePrivilege(context);
+        }
+
+        @Override
+        public Void visitShowProcStmt(ShowProcStmt statement, ConnectContext context) {
+            return checkStmtOperatePrivilege(context);
+        }
+
+        @Override
+        public Void visitSetStatement(SetStmt statement, ConnectContext context) {
+            List<SetVar> varList = statement.getSetVars();
+            varList.forEach(setVer -> {
+                if ((setVer instanceof SetPassVar)) {
+                    UserIdentity prepareChangeUser = ((SetPassVar) setVer).getUserIdent();
+                    try {
+                        prepareChangeUser.analyze();
+                    } catch (AnalysisException e) {
+                        ErrorReport.reportSemanticException(ErrorCode.ERR_UNKNOWN_ERROR, "ANALYZE ERROR");
+                    }
+                    if (!context.getUserIdentity().equals(prepareChangeUser)) {
+                        if (!PrivilegeManager.checkSystemAction(context, PrivilegeType.SystemAction.GRANT)) {
+                            ErrorReport.reportSemanticException(ErrorCode.ERR_SPECIFIC_ACCESS_DENIED_ERROR, "GRANT");
+                        }
+                    }
+                    return;
+                }
+                if (setVer.getType().equals(SetType.GLOBAL)) {
+                    checkStmtOperatePrivilege(context);
+                }
+            });
+            return null;
+        }
+
+        // ---------------------------------------- Common Privilege logic -------------------------------------------------
+        private Void checkStmtOperatePrivilege(ConnectContext context) {
```

Comment:
plz put the check method with other check methods in this file ,like `checkDbAction` etc.

Reply:
done